akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala

```diff
@@ -291,6 +295,81 @@ class CoordinatedShutdownSpec extends AkkaSpec {
       system.whenTerminated.isCompleted should ===(true)
     }
 
+    // exercising all the code paths through the terminate shotdown phase
+    "add and remove user JVM hooks 1" in {
```

Comment:
Test body could be factored out to a separate method and then reused in all of the three testcases.

Also testcase names could mention coordinated-shutdown parameters they exercise instead of number. :)